modules/styleguide/assets/js/frontend/contexts/settings.js

```diff
@@ -0,0 +1,164 @@
+import { createContext, useContext, useEffect, useState, useRef } from 'react';
+import useDebouncedCallback from '../hooks/use-debounced-callback';
+
+const SettingsContext = createContext( null );
+
+/**
+ * @return {{settings: Map, isReady: boolean}|null} context
+ */
+export const useSettings = () => {
+	return useContext( SettingsContext );
+};
+
+export const SettingsProvider = ( props ) => {
+	const [ status, setStatus ] = useState( 'idle' );
+	const [ settings, _setSettings ] = useState( new Map() );
+	const settingsRef = useRef( settings );
+
+	const setSettings = ( newSettings ) => {
+		settingsRef.current = newSettings;
+		_setSettings( newSettings );
+	};
+
+	useEffect( () => {
+		setStatus( 'loaded' );
+	}, [ settings ] );
+
+	const getInitialSettings = () => {
+		setStatus( 'loading' );
+
+		const kitSettings = elementor.config.document.settings.settings;
+
+		var map = new Map( [
+			[ 'colors', new Map( [
+				[ 'system_colors', kitSettings.system_colors ],
+				[ 'custom_colors', kitSettings.custom_colors ],
+			] ) ],
+			[ 'fonts', new Map( [
+				[ 'system_typography', kitSettings.system_typography ],
+				[ 'custom_typography', kitSettings.custom_typography ],
+				[ 'fallback_font', kitSettings.default_generic_fonts ],
+			] ) ],
+			[ 'config', new Map( [
+				[ 'is_debug', elementorCommon.config.isElementorDebug ],
+			] ) ],
+		] );
+
+		setSettings( map );
+	};
+
+	const onCommandEvent = ( event ) => {
+		switch ( event.detail.command ) {
+			case 'document/elements/settings':
+				onSettingsChange( event.detail.args );
+				break;
+			case 'document/repeater/insert':
+				onInsert( event.detail.args );
+				break;
+			case 'document/repeater/remove':
+				onRemove( event.detail.args );
+				break;
```

Comment:
How do you know that the repeater is `colors` or `typography`

Reply:
Maybe try to get the last snapshot of the data instead of trying to calculate the delta?